python/lsst/ip/isr/isrTask.py

```diff
@@ -839,7 +839,8 @@ def adaptArgsAndRun(self, inputData, inputDataIds, outputDataIds, butler):
         try:
             inputData['detectorNum'] = int(inputDataIds['ccdExposure']['detector'])
         except Exception as e:
-            raise ValueError(f"Failure to find valid detectorNum value for Dataset {inputDataIds}: {e}")
+            raise ValueError("Failure to find valid detectorNum value for Dataset %s: %s." %
+                             (inputDataIds, e))
```

Comment:
I think this is one instance where the `f`-string formatting would be OK. Don't mind either way.

Reply:
Do we have recommendations for when various methods are appropriate?  I have very little intuition on this topic.

Reply:
Perhaps @timj or @ktlim can make a suggestion?

Reply:
I don't understand why this needs to be moved back to `%` formatting as f-strings seem ideal for this one. I do have a preference for f-strings over `%` and `%` style is only required for logging.

Reply:
...and as I understand it, the “`%` style” which is required for logging is not the same as the above (I assume `"some string %s: %s" % (foo, bar)` suffers from the same defect as f-strings, in that it will always be interpolated by Python regardless of the log level).

I think the message here is, per https://developer.lsst.io/stack/logging.html#basic-usage-in-python, that when calling a logger you don't use Python string interpolation *at all*, but rather leave the logger to do the interpolation for you based on arguments you provide.

Reply:
Yes, you always defer the formatting when logging.